python/gstaichi/lang/_fast_caching/args_hasher.py

```diff
@@ -57,18 +60,21 @@ def stringify_obj_type(obj: Any) -> str | None:
     if isinstance(obj, MatrixField):
         return f"[fm-{obj.m}-{obj.n}-{obj.snode._id}-{obj.dtype}-{obj.shape}]"
     if dataclasses.is_dataclass(obj):
-        return dataclass_to_repr(obj)
+        return dataclass_to_repr(path, obj)
     if is_data_oriented(obj):
         child_repr_l = []
         for k, v in obj.__dict__.items():
-            _child_repr = stringify_obj_type(v)
+            _child_repr = stringify_obj_type(path + (k,), v)
             if _child_repr is None:
-                print("not representable child", k, type(v))
+                print("not representable child", k, type(v), "path", path)
                 return None
             child_repr_l.append(f"{k}: {_child_repr}")
         return ", ".join(child_repr_l)
-    if arg_type in [int, float, np.float32, np.float64, np.int32, np.int64, bool, np.bool]:
+    if arg_type in [int, float, np.float32, np.float64, np.int32, np.int64, bool]:
```

Comment:
`(int, float, np.float32, np.float64, np.int32, np.int64, bool)` ?

What about:
```python
import numbers
import numpy as np

if issubclass(arg_type, (numbers.Number, np.number)):
    [...]
```

Unless you want to exclude other numpy scalar types on purpose.